### PR TITLE
[heft] Improve UPGRADING.md note

### PR DESCRIPTION
**Pull request by octogonz**
We're preparing to release https://github.com/microsoft/rushstack/pull/2721

To view the formatted Markdown:  [UPGRADING.md](https://github.com/microsoft/rushstack/blob/octogonz/heft-upgrading-docs/apps/heft/UPGRADING.md)

### CODE REVIEW

File: apps/heft/UPGRADING.md [hunk 1]
```diff
@@ -2,41 +2,89 @@
...
+Breaking change for Jest:  This release of Heft enables rig support for Jest config files.
+It also reduces Heft's installation footprint by removing the Jest plugin from `@rushstack/heft`
+and moving it to its own package `@rushstack/heft-jest-plugin`.  As a result, Jest is now
+disabled by default.
+
+To reenable Jest support for your project, follow these steps:
+
+1. Add the `@rushstack/heft-jest-plugin` dependency to your project's **package.json** file.
+
+2. If you are using `@rushstack/heft-node-rig` or `@rushstack/heft-web-rig`, the Jest
```
Comment (D4N14L) [inline apps/heft/UPGRADING.md:14]: Maybe this should be the first step?

File: apps/heft/UPGRADING.md [hunk 2]
```diff
@@ -2,41 +2,89 @@
...
-    }
-  ]
-}
-```
+Breaking change for Jest:  This release of Heft enables rig support for Jest config files.
+It also reduces Heft's installation footprint by removing the Jest plugin from `@rushstack/heft`
+and moving it to its own package `@rushstack/heft-jest-plugin`.  As a result, Jest is now
+disabled by default.
+
+To reenable Jest support for your project, follow these steps:
```
Comment (D4N14L) [inline apps/heft/UPGRADING.md:10]: re-enable

File: apps/heft/UPGRADING.md [hunk 3]
```diff
@@ -2,41 +2,89 @@
...
+
+1. Add the `@rushstack/heft-jest-plugin` dependency to your project's **package.json** file.
+
+2. If you are using `@rushstack/heft-node-rig` or `@rushstack/heft-web-rig`, the Jest
+   plugin is already enabled.  Skip to step 4.
+
+3. Load the plugin by adding this setting to your `config/heft.json` file:
+
+   ```js
```
Comment (D4N14L) [inline apps/heft/UPGRADING.md:19]:
```json ?

**Review by D4N14L — APPROVED** [review]

File: apps/heft/UPGRADING.md (hunk 3 above)
Comment (octogonz) [reply apps/heft/UPGRADING.md:19]: This is how ` ```json ` handles comments:

```json
{
  // comment
}
```

Thus I always use ` ```js ` for JSON.

Comment (D4N14L) [reply apps/heft/UPGRADING.md:19]: Ahh that's interesting didn't know that. Makes sense.

File: apps/heft/UPGRADING.md (hunk 1 above)
Comment (octogonz) [reply apps/heft/UPGRADING.md]: Good feedack, fixed

File: apps/heft/UPGRADING.md (hunk 2 above)
Comment (octogonz) [reply apps/heft/UPGRADING.md:10]: Naw... In fact I dug my heels in and removed `re-include` from the Webpack note 😆

**Comment by octogonz** [comment]
The `@rushstack/heft-jest-plugin` package has been created, so we're ready to go whenever this PR is merged.

@iclanton @D4N14L

**Pull request merged**